xml/System.Collections.Concurrent/BlockingCollection`1.xml

```diff
@@ -1406,6 +1406,8 @@ This member is an explicit interface member implementation. It can be used only
         <exception cref="T:System.InvalidOperationException">The underlying collection was modified outside of this <see cref="T:System.Collections.Concurrent.BlockingCollection`1" /> instance, or the <see cref="T:System.Collections.Concurrent.BlockingCollection`1" /> is empty and the collection has been marked as complete for adding.</exception>
         <related type="Article" href="/dotnet/standard/collections/thread-safe/">Thread-Safe Collections</related>
         <related type="Article" href="/dotnet/standard/collections/thread-safe/blockingcollection-overview">BlockingCollection Overview</related>
+        <exception cref="T:System.OperationCanceledException">The <see cref="T:System.Collections.Concurrent.BlockingCollection`1" /> is empty and has been marked
+            as complete with regards to additions.</exception>
```

Comment:
This doesn't look right.  The docs correctly cite this same condition just a few lines above as being for an InvalidOperationException.  It's possible the docs were correctly updated and the source just never was? (I'm looking forward to a time when these are maintained in only one place.)

Reply:
I see what you mean. This is the text we have [in triple slash](https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L521-L530):

```cs
        /// <exception cref="System.OperationCanceledException">The <see
        /// cref="System.Collections.Concurrent.BlockingCollection{T}"/> is empty and has been marked
        /// as complete with regards to additions.</exception>
        /// <exception cref="System.ObjectDisposedException">The <see
        /// cref="System.Collections.Concurrent.BlockingCollection{T}"/> has been disposed.</exception>
        /// <exception cref="System.InvalidOperationException">The underlying collection was modified
        /// outside of this <see
        /// cref="System.Collections.Concurrent.BlockingCollection{T}"/> instance.</exception>
```

Should I revert this added exception? I am not convinced this API throws `OperationCanceledException` anywhere. If yes, then I think the triple slash exceptions should be updated as well.

Reply:
> Should I revert this added exception?

Yes, please.

> If yes, then I think the triple slash exceptions should be updated as well.

Yes, the /// comments are incorrect.  Not just on this overload but on others as well.

Reply:
It seems this is the only location that needs fixing in this PR. There's only [one other `Take` overload](https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L556), which takes a `CancellationToken` argument, and it can throw `OperationCanceledException` when it [internally calls](https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L649) `TryTakeWithNoTimeValidation`. It throws [here](https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L673) and [here](https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L719).

Reply:
> and it can throw OperationCanceledException when it [internally calls](https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L649) TryTakeWithNoTimeValidation. It throws [here](https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L673) and [here](https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L719).

Yes, but the docs on that overload are wrong:
https://github.com/dotnet/runtime/blame/60f593b4dc3760268956b9a874daad7f49b0255d/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs#L546-L549
It can throw an OperationCanceledException, but not because the collection "is empty and has been marked as complete with regards to additions."

Reply:
Opened issue to track the triple slash removal: https://github.com/dotnet/runtime/issues/66374